docs/source/command_line.rst

```diff
@@ -482,6 +482,32 @@ of the above sections.
            # 'items' now has type List[List[str]]
            ...
 
+.. option:: --local-partial-types
+
+    By default, mypy won't check partial types spanning module top level or class top level.
+    This flag changes the behavior to only allow partial types at local level. The most common
+    cases are variables initialized with values such as ``[]``, ``{}`` where the explicit type
+    arguments ``T`` for ``List[T]``, ``Dict[T]`` is missing, or ``None`` without explicit
+    ``Option`` annotation, therefore their types can not be fully inferred and specified. For example:
+
+    .. code-block:: python
+
+        from typing import Optional, List, Dict
+
+        a = None  # error
+        b = None  # type: Optional[int]
+        l1 = []   # error
+        l2 = []   # type: List[int]
```

Comment:
The list and dict examples are not relevant, since this option has no effect on them.